### DISCUSSION TITLE
Issue: Setting-up dedicated PlantUML server for wiki.js

### DISCUSSION THREAD

**Discussion by Congenial-GmbH** [requarks/wiki#7258] [Help / Questions]
I am running wiki.js using Ubuntu and Docker according to the [official Ubuntu setup-guide](https://docs.requarks.io/install/ubuntu) and would like to run my own PlantUML server on the same instance as described in [Quick Start Guide to PlantUML](https://plantuml.com/starting#:~:text=Access%20PlantUML%20Server,visual%20representation%20on%20the%20fly.):

**Steps taken:**

1. Set up a new wiki.js instance acc. to [official Ubuntu setup-guide](https://docs.requarks.io/install/ubuntu) .
2. Set up a PlantUML server using Docker:
    ```bash
    docker run -d -p 8080:8080 plantuml/plantuml-server:jetty
    ```
3. Configured Wiki.js to use the PlantUML server:
-   wiki.js: Administration ▸ Rendering ▸ PlantUML
-   Field: "PlantUML Server" 
-   Populated value: `http://localhost:8080` (or `http://localhost:8080/plantuml`)

4. Created a PlantUML diagram in the Markdown editor:
   ````
   ```plantuml
   @startuml
   Bob -> Alice : hello
   @enduml
   ```
   ````

**Issue**:
The diagram renders correctly in the edit mode preview but shows 'uml diagram' instead of the rendered diagram after saving and exiting the page.

**Additional Notes**:
- Docker container for PlantUML is running (acc. to `docker ps`)
- PlantUML service can be reached from browser at `http://<ip-address>:8080` and seems to work fine.
(Note: I will set-up SSL certifiacte later).

**Comment by Congenial-GmbH** [comment]
I made the following additional observations:

Very simple diagrams such as:
````
```plantuml
digraph G {
   A -> B
}
```
````
are rendered correctly when I exit the markdown editor.

More complex ones fail, and they also fail on the web-frontend `http://<ip-address>:8080` unless I adjust Settings **"Editor Watcher Timeout"** to a large number, such as `5000`.

I therefore suppose, I need to run my docker container with adjusted options as described at [How to set PlantUML options](https://github.com/plantuml/plantuml-server?tab=readme-ov-file#how-to-set-plantuml-options), e.g.

```bash
docker run -d -p 8080:8080 -e HTTP_PROXY_READ_TIMEOUT=20000 plantuml/plantuml-server:jetty
```

However, this time-out does not seem to be the reason why PlantUML rendering fails when exiting the markdown editor in wiki.js.

Does anyone have an idea how to resolve this issue?

**Comment by CleyFaye** [comment]
Note that configuring the plantuml rendering server in the wikijs admin have no effect on the editor preview, as requests will be sent to plantuml.requarks.io anyway. So, the editor preview working is no indication that your local setup works.

When the markdown page is rendered, it creates an img tag that will reference your configured plantuml render server from the client; meaning that if you configured your plantuml render server to be "http://localhost:8080", the client will try to connect to localhost when the page is displayed. This may or may not be what you want.

As far as I know, there is no option for wikijs to generate the image, store it, and serve it instead of linking to the plantuml render server.

**Comment by igoriok1994** [comment]
Any updates on Wiki.js 2.5.306?
It would be great to have this feature to set dedicated PlantUML / Mermaid / Kroki server.

**Comment by iskrenpp** [comment]
Custom PlantUML server worked for me in wikijs 2.5.307. The issue for the fix with umldiagram error on page rendering was that since my wikijs was behind https and I was trying to call local plantuml server with http - the mixed content browser error was preventing for proper comms.

As soon as I configured DNS CNAME and pointed to my local nginx-pm proxy with SSL then the local PlantUML url with `https://<CNAME> `started working with no issue in page edit mode rendering preview and also on page display.

Now finally I can use PlantUML server that is not 1000+ days old ( https://plantuml.requarks.io ) and has more recent features support
